### ISSUE TITLE
Support for fragments

### ISSUE THREAD

**Issue by manas-chaudhari** [manas-chaudhari/android-mvvm#43]
`MvvmActivity` allows for easy setup of activities.
A similar `MvvmFragment` can be added to allow easy setup of fragments.

**Comment by vinayppatil1989** [comment]
@manas-chaudhari How do we add/remove fragments from VM? Since adding fragments requires a TAG, LayoutId and FragmentManager a service won't do since that would require a separate method for every fragment(because layoutID can't be passed from VM or can we?). Your thoughts please.

**Comment by manas-chaudhari** [comment]
@vinayppatil1989 You're right. VM shouldn't be aware about `layoutId`. You can pass an interface to the View similar to `Navigator` in the sample and perform the fragment transaction in its implementation. That way, the fragment logic stays in your view layer (parent fragment/activity)

**Comment by vinayppatil1989** [comment]
@manas-chaudhari That way, wouldn't the interface impl get bloated up with a method for every fragment. It gets worse with fragment removing logic.

**Comment by manas-chaudhari** [comment]
Yes, it would. But, I think its necessary if you have different parameters for different navigations. If you have common arguments for navigation, you can pass the information as a parameter. For example, you could define a `Page` enum (or an `IntDef`) and pass it to the navigation method. The activity/fragment can store a mapping from Page -> layoutIds.

**Comment by vinayppatil1989** [comment]
Got that. But before I take that approach, what problems do you see in using R files in VM's?

**Comment by manas-chaudhari** [comment]
Problem is that that would couple your layout to the ViewModel. Hence, you won't be able to use the same ViewModel with another layout.
